Sources/PackageGraph/Pubgrub.swift

```diff
@@ -493,17 +567,21 @@ public final class PubgrubDependencyResolver<
     /// refer to. This means an incompatibility can occur several times.
     var incompatibilities: [Identifier: [Incompatibility<Identifier>]] = [:]
 
-    /// The root package reference.
-    var root: Identifier? {
-        didSet {
-            guard let root = root else { return }
-            // .unversioned might be a better case for the root package, but .versionSet(.any)
-            // ensures that no special handling is necessary for unit propagation
-            // when checking the solution's satisfiability against the root package.
-            add(Incompatibility(Term(not: root, .versionSet(.any)), cause: .root))
+    /// Find all incompatibilities containing a positive term for a given package.
+    func positiveIncompatibilities(for package: Identifier) -> [Incompatibility<Identifier>]? {
+        guard let all = incompatibilities[package] else {
+            return nil
+        }
+        return all.filter {
+            $0.terms.contains { term in
+                term.package == package && term.isPositive == true
```

Comment:
Isn't `term.package == package` always true since incompatibilities are mapped by the `package`?

Reply:
If yes, we should probably assert that instead of comparing.

Reply:
Aha right. I think it might be helpful to rename `all` to `packageIncompatibilities` or something.

Reply:
I chose to rewrite that statement in a more understandable way instead. Contains doesn't really carry the semantic meaning I was going for here well.